### PR TITLE
UI/managed namespace changes

### PR DESCRIPTION
**Pull request by chelshaw**
These changes include: 

* un-editable namespace prefix on auth route, if managed
* redirect to base namespace (for logged in user) or managed namespace root if no namespace param is present

- [x] Tests

**When a root namespace is retrieved from endpoint and no namespace param is present, redirect:**
rootNamespaceRedirect

### CODE REVIEW

File: ui/app/controllers/vault/cluster/auth.js [hunk 1]
```diff
@@ -7,11 +7,32 @@ export default Controller.extend({
...
   authMethod: '',
   redirectTo: alias('vaultController.redirectTo'),
+  managedNamespaceRoot: alias('configService.managedNamespaceRoot'),
+
+  get managedNamespaceChild() {
+    let fullParam = this.namespaceQueryParam;
+    let split = fullParam.split('/');
+    if (split.length > 1) {
+      split.shift();
+      console.log(split);
```
Comment (Monkeychip) [inline ui/app/controllers/vault/cluster/auth.js:23]: just heads up that there's a console log here

File: ui/app/controllers/vault/cluster/auth.js [hunk 2]
```diff
@@ -7,11 +7,32 @@ export default Controller.extend({
...
+      return `/${split.join('/')}`;
+    }
+    return '';
+  },
+
+  updateManagedNamespace: task(function*(value) {
+    yield timeout(500);
```
Comment (Monkeychip) [inline ui/app/controllers/vault/cluster/auth.js:30]: curious about the need for a timeout here?  maybe add a comment

File: ui/app/services/config.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import Service from '@ember/service';
+
+export default Service.extend({
```
Comment (Monkeychip) [inline ui/app/services/config.js:3]: With octane you should be able to start writing these as Native classes (however, ignore me if we have to backport to prior to 1.7). 
ex: 
```
import Service from '@ember/service';

export default class ShoppingCartService extends Service {
}
```

**Review by Monkeychip — COMMENTED** [review]
nice work so far!

File: ui/app/services/config.js (hunk 3 above)
Comment (chelshaw) [reply ui/app/services/config.js:3]: Great catch, however we do indeed have to backport this to prior to 1.7 😭

File: ui/app/controllers/vault/cluster/auth.js (hunk 2 above)
Comment (chelshaw) [reply ui/app/controllers/vault/cluster/auth.js:30]: Yes, same reason for the timeout on the manageNamespace fn but I'll copy it over. 

I could also make this DRYer by refactoring the original method but I wanted to not touch as much of the original functioning code as possible. Thoughts?

Comment (Monkeychip) [reply ui/app/controllers/vault/cluster/auth.js:30]: Maybe just add a comment. I agree, that maybe not touch the original function.

**Review by Monkeychip — APPROVED** [review]
Nice work!

File: ui/mirage/config.js [hunk 4]
```diff
@@ -2,6 +2,7 @@ export default function() {
   this.namespace = 'v1';
 
   this.get('sys/internal/counters/activity', function(db) {
+    console.log('getting sys/internal/counters/activity');
```
Comment (chelshaw) [inline ui/mirage/config.js:5]: Remove logs

**Pull request merged**